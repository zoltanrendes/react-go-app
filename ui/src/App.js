import React, { useState, useEffect } from "react";
import axios from "axios";

import logo from "./logo.svg";
import "./App.css";

function App() {
    const [state, setState] = useState([]);

    useEffect(() => {
        axios
            .get("/main")
            .then((resp) => {
                setState(resp.data)
            })
            .catch((err) => console.log('Error', err));
    }, []);

    return (
        <div className="App">
            <header className="App-header">
                <img src={logo} className="App-logo" alt="logo" />
                <p>Hello, {state.text}!</p>
            </header>
        </div>
    );
}

export default App;
