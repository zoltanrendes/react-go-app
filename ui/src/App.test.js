import React from "react";
import { mount, shallow } from "enzyme";
import { act } from "react-dom/test-utils";

import App from "./App";

const axios = require("axios");
const MockAdapter = require("axios-mock-adapter");
const mock = new MockAdapter(axios);

mock.onGet("/hello").reply(200, { text: "World" });

async function updateWrapper(wrapper, amount = 0) {
    await act(async () => {
        wrapper.update();
    });
}

describe("App.js", () => {
    beforeAll(() => jest.spyOn(React, "useEffect").mockImplementation(React.useLayoutEffect));
    afterAll(() => React.useEffect.mockRestore());
    it("it matches snapshot", async () => {
        await act(async () => {
            const wrapper = mount(<App />);
            expect(wrapper).toMatchSnapshot();
        });
    });

    it("returns data when sendMessage is called", async () => {
        let wrapper;

        await act(async () => {
            wrapper = mount(<App />);
        });

        const linkElement = wrapper.find("p");
        wrapper.update();
        expect(linkElement.text()).toBe('Hello, World!');
    });
});
