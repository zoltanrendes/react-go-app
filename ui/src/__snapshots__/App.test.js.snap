// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App.js it matches snapshot 1`] = `
<App>
  <div
    className="App"
  >
    <header
      className="App-header"
    >
      <img
        alt="logo"
        className="App-logo"
        src="logo.svg"
      />
      <p>
        Hello, 
        !
      </p>
    </header>
  </div>
</App>
`;
